compressed_image_transport/include/compressed_image_transport/compressed_subscriber.h

```diff
@@ -61,11 +64,14 @@ class CompressedSubscriber : public image_transport::SimpleSubscriberPlugin<sens
   virtual void internalCallback(const sensor_msgs::CompressedImageConstPtr& message,
                                 const Callback& user_cb);
 
+  sensor_msgs::ImagePtr decompressJPEG(const std::vector<uint8_t>& data, const std::string& source_encoding, const std_msgs::Header& header);
+
   typedef compressed_image_transport::CompressedSubscriberConfig Config;
   typedef dynamic_reconfigure::Server<Config> ReconfigureServer;
   boost::shared_ptr<ReconfigureServer> reconfigure_server_;
   Config config_;
   int imdecode_flag_;
+  tjhandle tj_;
```

Comment:
I tested performance without the class member (initing decompressor in each callback) and the performance loss wasn't that big. So if we'd want to keep ABI, this could be also a possibility.

Reply:
I guess that is up to you. I still think nobody is using `compressed_image_transport` directly, so there should be no impact of an ABI change. But I'm also fine with moving it to the callback.

Reply:
Feel free to leave it as it is. I think ABI compatibility here is not a key aspect as we're talking about a plugin.